### DISCUSSION TITLE
Compiling against musl target with openssl-sys dependencies

### DISCUSSION THREAD

**Discussion by tylerjl** [ipetkov/crane#454] [Q&A]
Hey hey! I'm running into issues trying to statically compile against a musl target that has `openssl-sys` as part of its dependency tree.

I've set the suggested flags and options per the helpful examples, so I'm halfway there and know that cargo is _trying_ to make a static compilation happen:

```
                  CARGO_BUILD_TARGET = "x86_64-unknown-linux-musl";
                  CARGO_BUILD_RUSTFLAGS = "-C target-feature=+crt-static";
```

...and my toolchain has musl as an additional target. The issue is that when it comes time to build the `openssl-sys` crate (which is deeper in my dependency tree as a dependency of another crate), I'm stuck at the following error:

```
       > error: could not find native static library `ssl`, perhaps an -L flag is missing?
       >
       > error: could not compile `openssl-sys` (lib) due to previous error
```

I suppose this error makes sense - the crate is trying to find a static library for `ssl` and not finding it. Per the typical way to do this, I previously had the _normal_ versions of `openssl` and `pkg-config` listed as inputs to the derivation, but in an effort to _provide_ the static `ssl` library, the following doesn't seem to help either:

```nix
let
  openssl = pkgs.pkgsMusl.openssl;
  pkg-config = pkgs.pkgsMusl.pkg-config;
in
{
  # ... other crane directives before this
  OPENSSL_STATIC = 1;
  OPENSSL_DIR = openssl;
  OPENSSL_LIB_DIR = openssl;
  OPENSSL_INCLUDE_DIR = openssl;
  buildInputs = [ makeWrapper openssl pkg-config ];
}
```

Any ideas? I've also tried doing weird stuff like instantiating my crane lib with something like `(crane.mkLib pkgs.pkgsMusl)` but that doesn't seem to do it either.

This may be something more related to nix and cargo in general, but I'm hoping that there may be something here that I may be overlooking.

**Comment by ipetkov** [comment]
Hi @tylerjl! Normally I'd expect to see `pkg-config` (and probably `makeWrapper`) listed under `nativeBuildInputs`; maybe that's the issue?
